src/map/script.c

```diff
@@ -16515,6 +16515,53 @@ static BUILDIN(autoequip)
 	return true;
 }
 
+/**
+ * equipidx(<index>{, <item id>})
+ */
+static BUILDIN(equipidx)
+{
+	int nameid = 0, i = script_getnum(st, 2);
+	struct item_data *item_data;
+	struct map_session_data *sd = script->rid2sd(st);
+
+	if (sd == NULL) {
+		script_pushint(st, 0);
+		return true;
+	}
+
+	if (i < 0 || i >= sd->status.inventorySize) {
+		ShowError("buildin_equipidx: Index (%d) should be from 0-%d.\n", i, sd->status.inventorySize - 1);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	if (i < sd->status.inventorySize) {
+		if (script_hasdata(st, 3))
+			nameid = script_getnum(st, 3);
+		else
+			nameid = sd->status.inventory[i].nameid == nameid;
```

Comment:
here look like error. Probably you want left only code ``nameid = sd->status.inventory[i].nameid``